ReLERNN/manager.py

```diff
@@ -0,0 +1,369 @@
+            win_len = win[2]
+            win_ct = win[6]
+            start = 0
+            for i in range(win_ct):
+                genomic_wins.append([win_chrom, start, win_len])
+                start += win_len
+
+        # partition for multiprocessing
+        mpID = range(len(genomic_wins))
+        task_q = mp.JoinableQueue()
+        result_q = mp.Queue()
+        params=genomic_wins, mask, maxLen
+
+        # do the work
+        pids = self.create_procs_maskWins(nProc, task_q, result_q, params)
+        self.assign_task_maskWins(mpID, task_q, nProc)
+        try:
+            task_q.join()
+        except KeyboardInterrupt:
+            print("KeyboardInterrupt")
+            sys.exit(0)
+
+        masks = []
+        for i in range(result_q.qsize()):
+            item = result_q.get()
+            masks.append(item)
+
+        mask_fraction, win_masks = [], []
+        for mask in masks:
+            mask_fraction.append(mask[0])
+            win_masks.append(mask)
+
+        print("{}% of genome inaccessible".format(round(self.mean(mask_fraction)*100,1)))
+        return self.mean(mask_fraction), win_masks
+
+
+    def maskStats(self, wins, last_win, mask, maxLen):
+        """
+        return a three-element list with the first element being the total proportion of the window that is masked,
+        the second element being a list of masked positions that are relative to the windown start=0 and the window end = window length,
+        and the third being the last window before breaking to expidite the next loop
+        """
+        chrom = wins[0].split(":")[0]
+        a = wins[1]
+        L = wins[2]
+        b = a + L
+        prop = [0.0,[],0]
+        try:
+            for i in range(last_win, len(mask[chrom])):
+                x, y = mask[chrom][i][0], mask[chrom][i][1]
+                if y < a:
+                    continue
+                if b < x:
+                    return prop
+                else:  # i.e. [a--b] and [x--y] overlap
+                    if a >= x and b <= y:
+                        return [1.0, [[0,maxLen]], i]
+                    elif a >= x and b > y:
+                        win_prop = (y-a)/float(b-a)
+                        prop[0] += win_prop
+                        prop[1].append([0,int(win_prop * maxLen)])
+                        prop[2] = i
+                    elif b <= y and a < x:
+                        win_prop = (b-x)/float(b-a)
+                        prop[0] += win_prop
+                        prop[1].append([int((1-win_prop)*maxLen),maxLen])
+                        prop[2] = i
+                    else:
+                        win_prop = (y-x)/float(b-a)
+                        prop[0] += win_prop
+                        prop[1].append([int(((x-a)/float(b-a))*maxLen), int(((y-a)/float(b-a))*maxLen)])
+                        prop[2] = i
+            return prop
+        except KeyError:
+            return prop
+
+
+    def mean(self, L):
```

Comment:
Is this function necessary? and if so, why make it a method instead of a helper function. While OOP is great, methods should only apply to instances of an object. This generally means that if it is not referencing "self" in some way in its code, it probably doesn't need to be a method - as opposed to a general function.